Dynatrace/dynatrace-operator
# Fixes an issue where the ActiveGate's statefulset is deleted and recreated if labeled manually

# Description

The activegate's default labels where changed some time ago. The activegate reconciler checks the labels of the deployed activegate against the desired one to update the statefulset, i one with old labels is deployed. This caused the activegate statefulset to be deleted and re-created if it was labeled by a third party, e.g. some webhook that automatically labels new resources. These changes modify that logic so only the old labels are checked and the statefulset is only deleted if they specifically exist.

## How can this be tested?

* Deploy the DTO and dynakube with AG enabled
* Wait until the statefulset is deployed
* Label the statefulset manually
* Make sure the statefulset is not deleted
* Change the activegate section of the dynakube
* Make sure the statefulset is still recreated correctly
* Label the statefulset manually
* Make sure the statefulset is not deleted


## Checklist
